### PR TITLE
Updated the docs for radiopalette

### PR DESCRIPTION
**Pull request by Daksh**

### CODE REVIEW

File: src/sugar3/graphics/radiopalette.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Please add a short description about the top of the file and embed your example code.

File: src/sugar3/graphics/radiopalette.py [hunk 2]
```diff
@@ -22,6 +22,10 @@
 
 
 class RadioMenuButton(ToolButton):
+    '''
+    This is a button used to open a group of widgets in a RadioPalette(menu)
+    It is not clickable
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: What are the kwargs that I should pass to it?  Maybe give a 1 line example of construction?

Also, what does it do?  You say it opens a group of buttons (what does this mean?) but is not clickable (my mind is very confused).

File: src/sugar3/graphics/radiopalette.py [hunk 3]
```diff
@@ -36,24 +40,38 @@ def __init__(self, **kwargs):
         self.connect('notify::palette', self.__palette_cb)
 
     def __palette_cb(self, widget, pspec):
+        '''
+        A callback function
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Implimentation method?

File: src/sugar3/graphics/radiopalette.py [hunk 4]
```diff
@@ -36,24 +40,38 @@ def __init__(self, **kwargs):
...
+    This is a button used to open a group of widgets in a RadioPalette(menu)
+    It is clickable
+    '''
 
     def __init__(self, **kwargs):
         RadioMenuButton.__init__(self, **kwargs)
 
     def do_clicked(self):
+        '''
+        This function is called when RadioToolsButton is clicked
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Implimentation method?  The API user should never call this right?

File: src/sugar3/graphics/radiopalette.py [hunk 5]
```diff
@@ -36,24 +40,38 @@ def __init__(self, **kwargs):
...
+        '''
         if not isinstance(self.props.palette, RadioPalette):
             return
         self.props.palette.update_button()
 
 
 class RadioToolsButton(RadioMenuButton):
+    '''
+    This is a button used to open a group of widgets in a RadioPalette(menu)
+    It is clickable
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Same comments as above.  How do I construct one?  What is it actually?

File: src/sugar3/graphics/radiopalette.py [hunk 6]
```diff
@@ -36,24 +40,38 @@ def __init__(self, **kwargs):
...
             return
         self.selected_button.emit('clicked')
 
 
 class RadioPalette(Palette):
-
+    '''
+    This contains the widgets to be displayed in the RadioMenuButton
+    and RadioToolsButton. This is the Palette (menu) that shows up when
+    those buttons are clicked. It is used to display the widgets.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Great!  But how does this differ from a normal palette?

Also, a usage example (just copy and paste one from your big example script) would help tons.

File: src/sugar3/graphics/radiopalette.py [hunk 7]
```diff
@@ -62,6 +80,9 @@ def __init__(self, **kwargs):
         self.set_content(self.button_box)
 
     def append(self, button, label):
+        '''
+        Adds a widget to the RadioPalette
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Please document the parameters, as per the example patch included in the gci task.

Also please document how this function ".show()s" the button, deals with it being clicked (it does right?) and says the button must not have subpalettes.  The activity developer should understand what this does using only your peices of text (and not seeing the code).

File: src/sugar3/graphics/radiopalette.py [hunk 8]
```diff
@@ -76,10 +97,17 @@ def append(self, button, label):
             self.__clicked_cb(button)
 
     def update_button(self):
+        '''
+        Updates the button
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: What does this mean?  Does it spin it in a circle?  Does it change everything to use cat pictures?

File: src/sugar3/graphics/radiopalette.py [hunk 9]
```diff
@@ -76,10 +97,17 @@ def append(self, button, label):
...
+        '''
         for i in self.button_box.get_children():
             self.__clicked_cb(i)
 
     def __clicked_cb(self, button):
+        '''
+        Called when an element in RadioPalette is clicked 
+        Changes the label text to reflect the change in the selection
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Implimentation method?

**Comment by samdroid-apps** [comment]
Good start, and the example works well!!.  But I have made some comments, please give 'em a look.

**Comment by Daksh** [comment]
@samdroid-apps I tried to fix all the suggestions that you gave, kindly have a look at it again?

File: src/sugar3/graphics/radiopalette.py [hunk 10]
```diff
@@ -36,24 +123,49 @@ def __init__(self, **kwargs):
...
         self.props.palette.update_button()
 
 
 class RadioToolsButton(RadioMenuButton):
+    '''
+    This is a button used to open a group of widgets in a RadioPalette(menu)
+    It is essentially just a RadioMenuButton but with the added function
+    do_clicked(self), which performs the given lines of code.    
+    Example Usage: button = RadioToolButton(icon_name='document-save',
+                                            group=group)
```
Comment (samdroid-apps) [inline src/sugar3/graphics/radiopalette.py]: Can you please document the icon_name and group as "Kwargs:" parameters?  Saying what type they must be will help readers.

**Pull request closed without merging**